session/session.go

```diff
@@ -1325,6 +1358,54 @@ func (s *session) Parse(ctx context.Context, sql string) ([]ast.StmtNode, error)
 	return stmts, nil
 }
 
+// ParseWithParams parses a query string, with arguments, to raw ast.StmtNode.
+func (s *session) ParseWithParams(ctx context.Context, sql string, args ...interface{}) ([]ast.StmtNode, error) {
+	sql, err := EscapeSQL(sql, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	internal := s.isInternal()
+
+	var stmts []ast.StmtNode
+	var warns []error
+	var parseStartTime time.Time
+	if internal {
+		// Do no respect the settings from clients, if it is for internal usage.
+		// Charsets from clients may give chance injections.
+		// Refer to https://stackoverflow.com/questions/5741187/sql-injection-that-gets-around-mysql-real-escape-string/12118602.
+		parseStartTime = time.Now()
+		stmts, warns, err = s.ParseSQL(ctx, sql, "utf8", "utf8_bin")
```

Comment:
I think this should be the 4 byte utf8 constants. i.e.:
```suggestion
		stmts, warns, err = s.ParseSQL(ctx, sql, mysql.DefaultCharset, mysql.DefaultCollationName)
```

Reply:
I used `utf8mb4` explicitly. I did not know that `utf8` is actually `utf8mb3`.